sdk/storage/storage-blob/test/perfstress/track-1/storageTest.spec.ts

```diff
@@ -0,0 +1,60 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { generateUuid } from "@azure/core-http";
+import {
+  Aborter,
+  ContainerURL,
+  ServiceURL,
+  SharedKeyCredential,
+  StorageURL
+} from "@azure/storage-blob";
+import { PerfStressTest, getEnvVar } from "@azure/test-utils-perfstress";
+
+// Expects the .env file at the same level as the "test" folder
+import * as dotenv from "dotenv";
+dotenv.config({ path: "../../../.env" });
+
+export abstract class StorageBlobTest<TOptions> extends PerfStressTest<TOptions> {
+  blobServiceClient: ServiceURL;
+  containerClient: ContainerURL;
+  static containerName = generateUuid();
+
+  constructor() {
+    super();
+    const connectionString = getEnvVar("STORAGE_CONNECTION_STRING");
+    const accountName = getValueInConnString(connectionString, "AccountName");
+    const accountKey = getValueInConnString(connectionString, "AccountKey");
+
+    const sharedKeyCredential = new SharedKeyCredential(accountName, accountKey);
+    this.blobServiceClient = new ServiceURL(
+      `https://${accountName}.blob.core.windows.net`,
+      StorageURL.newPipeline(sharedKeyCredential)
+    );
+    this.containerClient = ContainerURL.fromServiceURL(
+      this.blobServiceClient,
+      StorageBlobTest.containerName
+    );
+  }
+
+  public async globalSetup() {
+    await this.containerClient.create(Aborter.none);
```

Comment:
Call base setup and cleanup.

Reply:
This is the base class that extends `PerfStressTest`, in which `globalSetup` and the other methods aren't implemented.
Hence, not required.

And I think it would even run into an error for calling something that's not defined.

Reply:
You should always call the base methods, even if they are currently a no-op. It's the contract when overriding a method. It's possible we will add some logic to the based methods in the future.

Reply:
Yeah, what I was trying to say is it wouldn't be possible to use since it is not defined in the framework.
![image](https://user-images.githubusercontent.com/10452642/104053999-1b079b80-51a1-11eb-8fec-73ca355c6383.png)

To address your comment, I'll add no-op methods in the framework for future use which I could call in these base classes.

Reply:
Yes, please add no-op methods in the base class.  I'm not sure how it works in JS, but in .NET the difference is that `abstract` methods have no implementation in the base class and must be implemented by derived classes, while `virtual` methods must have an implementation (though it can be a no-op implementation).  So in .NET, the `setup/cleanup` methods are virtual with a no-op implementation, but the `run` methods are abstract which forces each test class to implement them or it would be a compile-time error.  The JS base class should be as similar as possible to .NET.

https://github.com/Azure/azure-sdk-for-net/blob/master/common/Perf/Azure.Test.Perf/PerfTest.cs

Reply:
Looks like TypeScript does support abstract methods, so `run()` and `runAsync()` should be abstract, while `setup/cleanup` should have no-op implementations.

https://www.typescriptlang.org/docs/handbook/classes.html#abstract-classes

Reply:
Here's a PR https://github.com/Azure/azure-sdk-for-js/pull/13125 to add the same.
Doing in a separate PR so that I don't have to duplicate in all the perf PRs.

Reply:
As per the conclusion from https://github.com/Azure/azure-sdk-for-js/pull/13125, we are retaining the current design of the framework which means we won't be calling the base methods in PerfStressTest class as they are not defined.